perf/reader.go

```diff
@@ -341,6 +340,30 @@ func (pr *Reader) Read() (Record, error) {
 	}
 }
 
+// Pause stops all notifications from this perf reader.
+//
+// Subsequent calls to Read will block until a call to Resume.
+func (pr *Reader) Pause() error {
```

Comment:
Sorry that I missed this earlier, but Pause and Resume have to take the mutex, since Close will modify the values.

Reply:
Turned out that this was a bit more involved than I initially thought, as the attempt to lock the mutex from a `Pause()`/`Resume()` call will block as long as `Read()` is blocking. So I needed to come up with an interrupt mechanism. Please take another look at this.

Reply:
Is Github acting up again? I can't see the interrupt mechanism in your PR.

Reply:
I ended up hacking away on it for a while but didn't come up with a solution that works reliably yet. I'll mention you directly on the main PR thread below when I post it, and make sure it's available for you to check out.

Reply:
Got it. I stared at the code for a bit as well, maybe the easiest solution is to just duplicate the raw fds into a separate array?

Reply:
So you mean have a distinct copy / distinct locks for the pause/resume? That may solve the issues I'm hitting, let me experiment..